Smart linking MVP: custom content type

orangejenny opened the pull request:
PR into https://github.com/dimagi/commcare-hq/pull/30002

https://dimagi-dev.atlassian.net/browse/USH-1186

orangejenny committed:
Removed unused users_extra views

orangejenny committed:
Removed admins_by_domain

orangejenny committed:
Replaced web_users_by_domain usage with dbaccessor

orangejenny committed:
Removed web_users_by_domain

joxl committed:
Add ES5 docker container configurations

Charl1996 committed:
Add coordinate to DHISFormConfig

Charl1996 committed:
Add geometry to enrollment

Charl1996 committed:
Test location parsing

Charl1996 committed:
Add support for feature type retrieval

Charl1996 committed:
Remove redundant comment

Charl1996 committed:
Use mobile user location as events/enrolments location reference

Charl1996 committed:
Fix str -> float issue

Charl1996 committed:
Fix geoJson type

Charl1996 committed:
Remove redundant constant

Charl1996 committed:
Fix accessor method

Charl1996 committed:
Move location extraction to form config

Charl1996 committed:
Update README

joxl committed:
Add ES5 python library and remove ES1

joxl committed:
Add ES5 compatibility and remove legacy ES1 code

joxl committed:
ES `retry_on_conflict` parameter is only valid with `update` requests

snopoke committed:
remove roles_by_domain couch view

minhaminha committed:
added checkbox function

minhaminha committed:
added checkbox function

minhaminha committed:
removed show field

minhaminha committed:
fixed double appearing form

minhaminha committed:
removed old code

mjriley committed:
always pass request to authenticate

Charl1996 committed:
Add new custom recipient option

minhaminha committed:
removed checkbox label

Charl1996 committed:
Fix merge conflict: leave label=''

Charl1996 committed:
Changes:

- Fix name typo

Add support for old reminders

Add test for new recipient

Charl1996 committed:
Rename confusing types

orangejenny committed:
Added build error for menu with session endpoint and display only forms

orangejenny committed:
Added test

orangejenny committed:
Updated default login as permission checkbox to be disabled when user can't login as

minhaminha committed:
raise 404 error when trying to access report while visibility is disabled

millerdev committed:
Remove docs for should_use_sql_backend

millerdev committed:
Couch backend out of corehq/apps/change_feed

millerdev committed:
Couch backend out of corehq/apps/cleanup

millerdev committed:
Couch backend out of corehq/apps/data_interfaces

millerdev committed:
Couch backend out of corehq/apps/commtrack

millerdev committed:
get_supply_point_ids_in_domain_by_location to SQL

millerdev committed:
isort

millerdev committed:
create_case still needs should_use_sql_backend

This commit moves create_case to a more logical place so all references to should_use_sql_backend can be removed from the data_interfaces app.

gherceg committed:
correct order of patched parameters

millerdev committed:
Couch backend out of corehq/apps/data_pipeline_audit

millerdev committed:
Fix test for USE_PARTITIONED_DATABASE=yes

Not sure how this test was ever passing in the past with USE_PARTITIONED_DATABASE=yes and a multi-shard config since AutomaticUpdateRule.iter_cases is called for each shard. Returning a list on a multi-shard config means the same case would be processed for each shard db. Returning an iterator causes it to be exhausted on the first shard, and all other shards to yield no cases.

millerdev committed:
Couch backend out of corehq/apps/hqadmin

gherceg committed:
define decorators once

millerdev committed:
Refactor: simplify republish_doc_changes

millerdev committed:
Remove unused import

orangejenny committed:
Update staging.yaml

orangejenny committed:
Merge pull request #30245 from dimagi/jls/endpoints-no-dof

Added build error for session endpoints to menus using display-only forms

joxl committed:
Update staging.yaml

joxl committed:
Update staging.yaml

orangejenny committed:
Fixed pact imports

mjriley committed:
Merge pull request #30225 from dimagi/mjr/record_failure_login_ips

minhaminha committed:
removed tests relating to Show filter

orangejenny committed:
Update staging.yaml

orangejenny committed:
Update staging.yaml

orangejenny committed:
Merge pull request #30275 from dimagi/jls/fix-pact-imports

Fixed pact imports

millerdev committed:
Do not use partitioned SQL database

Unlike @run_with_all_backends, tests that use @use_sql_backend are run with a _sharded_ SQL backend. Many tests will not work with a sharded backend.

millerdev committed:
Revert "Fix test for USE_PARTITIONED_DATABASE=yes"

This reverts commit 0e8f8158bd6f9d822734b49070fdfab521849f5c.

orangejenny committed:
Added custom content type to repsonse

orangejenny committed:
Update staging.yaml

esoergel committed:
updating files: scripts/staging.yaml

[ci skip]

snopoke commented on corehq/apps/cloudcare/views.py:
```diff
@@ -612,4 +612,6 @@ def _fail(error):
     })
-    return HttpResponseRedirect(reverse(FormplayerMain.urlname, args=[domain]) + "#" + cloudcare_state)
+    response = HttpResponseRedirect(reverse(FormplayerMain.urlname, args=[domain]) + "#" + cloudcare_state)
+    response['Content-Type'] = "application/commcare"
```
I think this should be a constant with a comment about it's purpose.

shubham1g5 committed:
updating files: scripts/staging.yaml

[ci skip]

shubham1g5 committed:
updating files: scripts/staging.yaml

[ci skip]

shubham1g5 commented:
@orangejenny I was trying out [this url](https://staging.commcarehq.org/a/grogu/app/v1/a208b08897ec464fbc48a8e80ed216c8/shows_endpoint/?case_id=b2bbfd8f-6c94-4d62-b7bf-1ef710653398&case_id_show=1bfb0ec9-174e-4f00-be30-25855747880a) to test but somehow couldnot get to open this link in CommCare on mobile. 

to debug, I did a `curl -v url` and I don't see the `application/commcare` value for the content-type in reponse headers -

```
< HTTP/2 302
< date: Tue, 24 Aug 2021 10:10:33 GMT
< content-type: text/html; charset=utf-8
< content-length: 0
< server: nginx
```

mkangia committed:
add staging branch

millerdev committed:
use_sql_backend -> use_sharded_db

use_sql_backend sets TESTS_SHOULD_USE_SQL_BACKEND in addition to adding a test attribute that causes Travis to run the tests on the sharded node. However, the TESTS_SHOULD_USE_SQL_BACKEND setting, which is also used by @run_with_all_backends does not itself imply use of a sharded database.

The words "sharded" and "partitioned" are used interchangeably throughout the codebase (see also USE_PARTITIONED_DATABASE). Another notable place where "sharded" is used is in the test runner directives "python-sharded" and "python-sharded-and-javascript".

millerdev committed:
Lint

snopoke committed:
view to get case XML from a data registry

millerdev committed:
sql_backend -> sharded_db

Rename attribute used to tag tests to run with sharded database.

millerdev committed:
Improve naming consistency and remove ambiguity

There is another unrelated and more commonly used @partitioned decorator used for database migrations.

millerdev committed:
@partitioned -> @use_sharded_db

joxl committed:
Merge pull request #30265 from dimagi/jm/es5-containers

Add ES5 containers to docker-compose configuration

joxl committed:
Update staging.yaml

millerdev committed:
Merge remote-tracking branch 'origin/master' into dm/sharded-db

millerdev committed:
@use_shard_db -> @sharded

orangejenny committed:
Merge pull request #30247 from dimagi/jls/checkbox-disabling

Updated default login as permission checkbox to be disabled when user…

millerdev committed:
Rename test tag: sharded_db -> sharded

orangejenny committed:
Update staging.yaml

snopoke committed:
tests for helper.get_case

snopoke committed:
differentiate between exceptions

snopoke committed:
add registry schema utility

snopoke committed:
lint

joxl committed:
Remove unused dev dependencies

- graphviz (previous usage unknown)
- recommonmark (usage removed in #30044)

dependabot[bot] committed:
Bump radon from 5.0.1 to 5.1.0 in /requirements

Bumps [radon](https://github.com/rubik/radon) from 5.0.1 to 5.1.0.
- [Release notes](https://github.com/rubik/radon/releases)
- [Changelog](https://github.com/rubik/radon/blob/master/CHANGELOG)
- [Commits](https://github.com/rubik/radon/compare/v5.0.1...v5.1.0)

---
updated-dependencies:
- dependency-name: radon
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

millerdev committed:
Replace use_sharded_db that were missed earlier

joxl committed:
Merge pull request #30268 from dimagi/dependabot/pip/requirements/radon-5.1.0

Bump radon from 5.0.1 to 5.1.0 in /requirements

joxl committed:
Merge pull request #30281 from dimagi/jm/requirements-cleanup

Remove unused dev dependencies

joxl committed:
Update staging.yaml

minhaminha committed:
cleaning up

mjriley committed:
Remove translation typo in enterprise views

mjriley committed:
Clean up block translation messages

biyeun committed:
ensure that we never invoice domains who are on a paused plan

biyeun committed:
add test for domain invoice factory with paused plans

biyeun committed:
ensure that customer accounts are never invoiced for paused plans

biyeun committed:
add test for customer invoice factory with paused plan

mjriley committed:
Customize makemessages to support ignoring fuzzy

mjriley committed:
Ignore obsolete and fuzzy translations

minhaminha committed:
Merge pull request #30273 from dimagi/ml/remove-show-field-sms-billables-report+ml/sms-billable-report-visibility-checkbox

Add checkbox toggle for SMS billable report visibility in the accounting UI + remove "show" filter in sms detailed reports

minhaminha committed:
Update staging.yaml

merged branches

joxl committed:
Merge pull request #30283 from dimagi/minhaminha-patch-2

Update staging.yaml

mjriley committed:
Updated staging.yaml

biyeun committed:
restore original behavior with community edition as DomainInvoiceFactory also relised on should_create_invoice and it is failing tests with this new change

gherceg committed:
consolidate custom multiselect bindings; rebuild instead of refresh

gherceg committed:
Update staging.yaml

gherceg committed:
Update staging.yaml

gherceg committed:
Update staging.yaml

gherceg committed:
Update staging.yaml

gherceg committed:
Merge pull request #30274 from dimagi/gh/fix-patch-order

Correct order of patched parameters

shubham1g5 committed:
updating files: scripts/staging.yaml

[ci skip]

dependabot[bot] committed:
Bump cryptography from 3.4.7 to 3.4.8 in /requirements

Bumps [cryptography](https://github.com/pyca/cryptography) from 3.4.7 to 3.4.8.
- [Release notes](https://github.com/pyca/cryptography/releases)
- [Changelog](https://github.com/pyca/cryptography/blob/main/CHANGELOG.rst)
- [Commits](https://github.com/pyca/cryptography/compare/3.4.7...3.4.8)

---
updated-dependencies:
- dependency-name: cryptography
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

millerdev committed:
Merge pull request #30279 from dimagi/dm/sharded-db

Clear up "SQL backend" / "sharded db" confusion

millerdev committed:
Merge remote-tracking branch 'origin/master' into dm/no-couch-backend-part1

snopoke committed:
rename view and url

snopoke committed:
disable remote request POST for data registry search

snopoke committed:
Merge branch 'master' into sk/ota-get-case

snopoke committed:
fix test with missing decorator

gherceg committed:
move comment

snopoke committed:
add 'query' to 'entry' for registry datums

snopoke committed:
fix circular import

joxl committed:
Merge pull request #30290 from dimagi/dependabot/pip/requirements/cryptography-3.4.8

Bump cryptography from 3.4.7 to 3.4.8 in /requirements

biyeun committed:
Merge pull request #30286 from dimagi/bmb/invoicing-paused-fix

[fix] do not invoice paused subscriptions with $0 invoices

gherceg committed:
Merge pull request #30289 from dimagi/gh/rebuild-multiselect

Fix multiselect knockout binding bug

mjriley committed:
Updated translation script to allow arguments

millerdev committed:
Merge pull request #30270 from dimagi/dm/no-couch-backend-part1

Remove Couch backend (part 1)

millerdev committed:
Improve cleanup in commtrack XML tests

millerdev committed:
Lint

millerdev committed:
Merge pull request #30284 from dimagi/mjr/translations_fix

Repair translations

millerdev committed:
Update translations.

[ci skip]

millerdev committed:
Update translations, removing fuzzy and obsolete

Command: ./scripts/update-translations.sh --no-obsolete --no-fuzzy

[ci skip]

millerdev committed:
Couch backend out of corehq/apps/hqcase

millerdev committed:
Couch backend out of corehq/apps/hqwebapp

orangejenny committed:
Update staging.yaml

millerdev committed:
Couch backend out of corehq/apps/receiverwrapper

mjriley committed:
Updates Swahili translations

mjriley committed:
Updated Hindi translations

mjriley committed:
Update French translations

mjriley committed:
Updated English Translations

orangejenny committed:
Updated feature flag page to sort items

mjriley committed:
Updated Spanish translations

orangejenny committed:
Fixed import and version in time_restore command

mjriley committed:
Updated Portuguese translations

millerdev committed:
Couch backend out of corehq/apps/reports

orangejenny committed:
Added quickcache to get_apps_in_domain

millerdev committed:
Couch backend out of corehq/apps/userreports

orangejenny committed:
Added cache clearing

orangejenny committed:
Update staging.yaml

snopoke committed:
Merge branch 'master' into sk/ota-get-case

snopoke committed:
Refactor ReportFixturesProvider to avoid loading apps and report configs twice

millerdev committed:
Couch backend out of corehq/ex-submodules/pillowtop

millerdev committed:
Couch backend out of corehq/messaging

millerdev committed:
Couch backend out of corehq/pillows

millerdev committed:
Couch backend out of corehq/reports.py

Also removed commtrack.ReportingRatesReport and commtrack.ReportingStatusMapReport and things only referenced by them.

millerdev committed:
Remove print [ci skip]

millerdev committed:
Merge pull request #30291 from dimagi/dm/no-couch-backend-part2

Remove Couch form processor: commtrack

snopoke committed:
call super.tearDownClass

millerdev committed:
Remove get_ledger_values_for_case_as_of

Also make SimplifiedInventoryDataSource use SQL and remove SimplifiedInventoryDataSourceNew.

orangejenny committed:
Merge pull request #30294 from dimagi/jls/fix-time-restore

Fixed import and version in time_restore command

orangejenny committed:
Merge pull request #30296 from dimagi/jls/cache-dbaccessor

Added quickcache to get_apps_in_domain

millerdev committed:
Run TestCaseSearchLookups tests with SQL backend

orangejenny committed:
Update staging.yaml

snopoke committed:
return True

snopoke committed:
Merge pull request #30280 from dimagi/sk/ota-get-case

OTA view to get case details from a data registry

millerdev committed:
Merge pull request #30292 from dimagi/dm/rm-couch-3

Remove Couch: data_pipeline_audit, hqadmin

Charl1996 committed:
updating files: scripts/staging.yaml

[ci skip]

millerdev committed:
Remove Couch-only StockState tests

snopoke committed:
Merge pull request #30298 from dimagi/sk/cache-apps-in-urc-fixture

Refactor ReportFixturesProvider to avoid loading apps and report configs twice

millerdev committed:
Fix TestFilterDslLookups and ElasticTestMixin

millerdev committed:
Disable Couch branch of run_with_all_backends

orangejenny committed:
Merge pull request #30293 from dimagi/jls/sort-flags

Updated feature flag page to sort items

millerdev committed:
Run more hqcase tests with SQL backend

mkangia committed:
remove superuser management from web user edit form

millerdev committed:
Fix tests with setUp(Class)? on SQL backend

orangejenny committed:
Switched from redirect response to final cloudcare response

mjriley committed:
Updated translations from transifex

millerdev committed:
Fix testapps tests on SQL backend

millerdev committed:
Fix couchforms.analytics tests

The code being tested by this module seems to be backend-agnostic, despite the name of the top-level package.

millerdev committed:
Fix corehq.apps.ota tests

mjriley committed:
Merge pull request #30295 from dimagi/mjr/translation_cleanup

snopoke committed:
record regisry data access

snopoke committed:
extract access checks

snopoke committed:
update tests

snopoke committed:
Merge branch 'master' into sk/remove-role-couch-views

snopoke committed:
Merge branch 'master' into jls/kill-phones-to-domains

snopoke committed:
update test case

snopoke committed:
fix imports

snopoke committed:
Merge pull request #30179 from dimagi/sk/remove-role-couch-views

remove roles_by_domain couch view

snopoke committed:
Merge pull request #29558 from dimagi/jls/kill-phones-to-domains

Removed web_users_by_domain couch view and a few unused views

snopoke committed:
docstring

Charl1996 committed:
Merge pull request #30237 from dimagi/cs/SC-1697_new-custom-recipient

Cs/sc 1697 new custom recipient

mkangia committed:
Merge pull request #30302 from dimagi/mk/remove-web-user-superuser

remove superuser management from web user edit form

millerdev committed:
Fix more test setups

millerdev committed:
Cherry-pick changes from dm/rm-couch-4

Single file picked from commit 18089c930a5080028b5f48369349fd216903aef8

Charl1996 committed:
Merge pull request #30016 from dimagi/cs/SC-691-add-support-for-geolocation

DHIS2: Add support for sending geolocation

AmitPhulera committed:
updating files: scripts/staging.yaml

[ci skip]

AmitPhulera committed:
Merge branch 'master' of https://github.com/dimagi/commcare-hq

snopoke committed:
Merge pull request #30310 from dimagi/sk/registry-search

data registry suite file changes for loading case data into forms

snopoke committed:
update registry helper tests

snopoke committed:
Merge branch 'master' into sk/registry-get-case

snopoke committed:
update registry_case url reference

shubham1g5 commented:
Thanks @orangejenny. I tried this out on mobile but this doesn't seem to work as well. This makes me think that we do need to switch to a path pattern to identify these urls. I will tag you on spec to discuss the same but we can close this one out. Thanks again for assiting with this.

snopoke committed:
fix test xml

snopoke committed:
add missing domain param

snopoke committed:
add missing domain param

millerdev committed:
Merge pull request #30297 from dimagi/dm/rm-couch-4

Remove Couch: hqcase, hqwebapp, receiverwrapper, reports, userreports

millerdev committed:
Merge pull request #30300 from dimagi/dm/rm-couch-6

Disable Couch branch of run_with_all_backends

millerdev committed:
Merge pull request #30299 from dimagi/dm/rm-couch-5

Remove Couch: pillowtop, corehq.messaging, corehq.pillows, corehq.reports

mkangia committed:
Merge pull request #30314 from dimagi/sk/fix-test

add missing domain param

snopoke committed:
Merge pull request #30305 from dimagi/sk/registry-get-case

log registry data access

snopoke committed:
updating files: scripts/staging.yaml

[ci skip]

snopoke committed:
Merge branch 'master' into jls/smart-linking-mvp-content-type
